airbyte-webapp/src/views/Connection/ConnectionForm/ConnectionForm.tsx

```diff
@@ -118,6 +118,19 @@ const DirtyChangeTracker: React.FC<DirtyChangeTrackerProps> = ({ dirty, onChange
   return null;
 };
 
+const fixScheduleData = (values: FormikConnectionFormValues) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const currentScheduleValue = values.scheduleData?.basicSchedule as any;
+
+  const isManualOrCron =
+    currentScheduleValue === ConnectionScheduleType.manual || currentScheduleValue === ConnectionScheduleType.cron;
+
+  return {
+    ...values.scheduleData,
+    basicSchedule: isManualOrCron ? undefined : (currentScheduleValue as ConnectionScheduleDataBasicSchedule),
+  };
+};
```

Comment:
Correct me if I'm wrong, but I think the problem this method is trying to solve is that if the user selects `Manual`, then this is trying to clear out the `values.scheduleData.basicSchedule` value, because that is currently being set to the string `"manual"` which is not a valid `ConnectionScheduleDataBasicSchedule` value.

Instead doing it this way, I think we should fix the root cause of why `values.scheduleData.basicSchedule` is being set to `"manual"` and/or `"cron"` in the first place, which we shouldn't do anyways since that is not a valid value to use for that property.

We should also fix this logic so that we are not leaving `values.scheduleData.cron` set to some value when a user goes from Cron -> Manual, which is currently the behavior.

See my comment below for some suggestions around this

Reply:
👍🏻